<a href="https://colab.research.google.com/github/charmingmichelle/DeepLearning_AW/blob/master/Intune_Tickets_04_23.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:
import warnings
warnings.filterwarnings('ignore')
import numpy as np
import pandas as pd
pd.set_option('display.max_columns', None)
import scipy.stats as stats
import matplotlib.pyplot as plt
%matplotlib inline
plt.rcParams['figure.figsize'] = (8, 5)
import seaborn as sns

# **Import Data**

In [4]:
from google.colab import files
uploaded = files.upload()

import io
my_data = pd.read_csv(io.BytesIO(uploaded['Intune_Tickets_4_23.csv']))

Saving Intune_Tickets_4_23.csv to Intune_Tickets_4_23.csv


In [5]:
my_data.head()

Number              Created              Caller User ID  \
0  INC27203974  2019-04-23 09:09:54     Alexander Carey  544788   
1  INC27203638  2019-04-23 08:55:57          Jane Blake  556147   
2  INC27202687  2019-04-23 08:08:50       Thomas Arthur  584025   
3  INC27202205  2019-04-23 07:28:59           Dan Hancu  599061   
4  INC27200879  2019-04-23 00:01:52  Sangeetha Umapathy  590667   

                                   Short description  \
0                              MDM | account blocked   
1                        tol mobile app not opening    
2               BAH apps not working Internationally   
3  received an email stating he is not in complia...   
4                    unable to access email on phone   

                                         Description  \
0  Employee ID #: 544788\nEmployee Name: Alexande...   
1  Employee ID #: 556147\nEmployee Name: Jane Bla...   
2  Employee ID #: 584025\nEmployee Name: Thomas A...   
3  Employee ID #: 599061\nEmployee Name: Dan Hanc...   
4  unable to access emails on phone\nadvised to i...   

          Configuration item       Support Function      Incident state  \
0          MDM-HD Resolvable  Security Policy Issue        Work Started   
1  Mobile Apps-HD Resolvable         Usage Question            Resolved   
2             MDM-Escalation         Usage Question     Awaiting Change   
3             MDM-Escalation         Usage Question  Awaiting User Info   
4          MDM-HD Resolvable         Usage Question            Resolved   

       Assignment group      Assigned to      Priority              Updated  \
0      BAH Service Desk   Sharon Johnson  3 - Moderate  2019-04-23 09:35:06   
1      BAH Service Desk  Chasity Ritchie  3 - Moderate  2019-04-23 08:59:30   
2  BAH BSO ESC Wireless              NaN  3 - Moderate  2019-04-23 08:51:50   
3  BAH BSO ESC Wireless       Carl Hecox  3 - Moderate  2019-04-23 08:45:39   
4      BAH Service Desk    Austin Yearta  3 - Moderate  2019-04-23 00:02:34   

  Parent             Resolved Additional comments  \
0    NaN                  NaN                 NaN   
1    NaN  2019-04-23 08:59:25                 NaN   
2    NaN                  NaN                 NaN   
3    NaN                  NaN                 NaN   
4    NaN  2019-04-23 00:02:34                 NaN   

                     Updated by        Opened by  \
0  Sharon.Johnson2@compucom.com   Sharon Johnson   
1    Chasity.Clark@compucom.com  Chasity Ritchie   
2  Curtis.Thompson@compucom.com  Curtis Thompson   
3                    bah\561695    Barnett Reece   
4    Austin.Yearta@compucom.com    Austin Yearta   

                     Created by  \
0  Sharon.Johnson2@compucom.com   
1    Chasity.Clark@compucom.com   
2  Curtis.Thompson@compucom.com   
3    Barnett.Reece@compucom.com   
4    Austin.Yearta@compucom.com   

                                          Work notes Contact type  
0                                                NaN        Phone  
1  2019-04-23 08:59:25 - Chasity Ritchie (Work no...        Phone  
2  2019-04-23 08:18:57 - Curtis Thompson (Work no...        Phone  
3  2019-04-23 08:45:39 - Carl Hecox (Work notes)\...        Phone  
4  2019-04-23 00:02:34 - Austin Yearta (Work note...        Phone

In [6]:
my_data.shape

(440, 21)

In [7]:
my_data.describe()

Number              Created            Caller User ID  \
count           440                  440               440     440   
unique          440                  440               407     407   
top     INC27197224  2019-04-16 11:58:19  Helene Berkholcs  579707   
freq              1                    1                 3       3   

                                    Short description  \
count                                             440   
unique                                            406   
top     I: Microsoft Authenticator questions - No ETR   
freq                                               12   

                                              Description Configuration item  \
count                                                 440                440   
unique                                                437                  5   
top     the client is being asked to setup the microso...  MDM-HD Resolvable   
freq                                                    3                249   

       Support Function Incident state  Assignment group Assigned to  \
count               440            440               440         434   
unique                8              7                 7          43   
top      Usage Question         Closed  BAH Service Desk  Carl Hecox   
freq                228            272               213          28   

            Priority              Updated       Parent             Resolved  \
count            440                  440          222                  338   
unique             2                  421            4                  337   
top     3 - Moderate  2019-04-23 02:14:17  INC27117897  2019-04-16 18:28:37   
freq             437                    3          212                    2   

                                      Additional comments Updated by  \
count                                                 356        440   
unique                                                347         47   
top     2019-04-22 02:18:35 - System (Additional comme...     system   
freq                                                    3        272   

          Opened by                Created by  \
count           438                       440   
unique          144                       145   
top     Craig Dupre  Craig.Dupre@compucom.com   
freq             24                        24   

                                               Work notes Contact type  
count                                                 416          440  
unique                                                416            6  
top     2019-04-22 08:01:29 - Jessica Robinson (Work n...        Phone  
freq                                                    1          309

# **Data Exploration**

**Priority**

In [0]:
priority = my_data.Priority

In [9]:
priority.describe()

count              440
unique               2
top       3 - Moderate
freq               437
Name: Priority, dtype: object

In [0]:
priority.head(10)

0    3 - Moderate
1    3 - Moderate
2    3 - Moderate
3    3 - Moderate
4    3 - Moderate
5    3 - Moderate
6    3 - Moderate
7    3 - Moderate
8    3 - Moderate
9    3 - Moderate
Name: Priority, dtype: object

In [0]:
my_data.groupby('Priority').describe()

Number                          Created         \
              count unique          top freq   count unique   
Priority                                                      
1 - Critical      3      3  INC27163383    1       3      3   
3 - Moderate    437    437  INC27156142    1     437    437   

                                       Caller                                \
                              top freq  count unique               top freq   
Priority                                                                      
1 - Critical  2019-04-17 14:47:43    1      3      3   Patricia Goshen    1   
3 - Moderate  2019-04-16 07:28:27    1    437    404  Helene Berkholcs    3   

             User ID                     Short description         \
               count unique     top freq             count unique   
Priority                                                            
1 - Critical       3      3  523297    1                 3      3   
3 - Moderate     437    404  579707    3               437    404   

                                                                      \
                                                            top freq   
Priority                                                               
1 - Critical  #1 MC Voicemail & Email 04/18/19 | Intune MAM ...    1   
3 - Moderate      I: Microsoft Authenticator questions - No ETR   12   

             Description         \
                   count unique   
Priority                          
1 - Critical           3      3   
3 - Moderate         437    434   

                                                                      \
                                                            top freq   
Priority                                                               
1 - Critical  forwarded by: laderer_beth@bah.com\n\nGood Mor...    1   
3 - Moderate  the client is being asked to setup the microso...    3   

             Configuration item                                 \
                          count unique                top freq   
Priority                                                         
1 - Critical                  3      2  MDM-HD Resolvable    2   
3 - Moderate                437      5  MDM-HD Resolvable  247   

             Support Function                             Incident state  \
                        count unique             top freq          count   
Priority                                                                   
1 - Critical                3      2  Usage Question    2              3   
3 - Moderate              437      8  Usage Question  226            437   

                                 Assignment group         \
             unique     top freq            count unique   
Priority                                                   
1 - Critical      1  Closed    3                3      3   
3 - Moderate      7  Closed  269              437      7   

                                                     Assigned to         \
                                            top freq       count unique   
Priority                                                                  
1 - Critical  BAH Service Desk Pending Customer    1           3      3   
3 - Moderate                   BAH Service Desk  212         431     43   

                               Updated                                   \
                      top freq   count unique                  top freq   
Priority                                                                  
1 - Critical  Mona Cahill    1       3      3  2019-04-20 02:19:38    1   
3 - Moderate   Carl Hecox   28     437    420  2019-04-23 02:14:17    3   

             Parent                          Resolved         \
              count unique          top freq    count unique   
Priority                                                       
1 - Critical      2      1  INC27117897    2        3      3   
3 - Moderate    220      4  INC27117

**Conclusion:** According to the analyst based on "Priority", there are：

*   437 cases are moderate(3)
*   3 casees are critical(1)



**Question:** Is there any other priorities? 2, 4, 5, etc?

**Support Function**

In [0]:
SF = my_data['Support Function']

In [15]:
SF.head()

0    Security Policy Issue
1           Usage Question
2           Usage Question
3           Usage Question
4           Usage Question
Name: Support Function, dtype: object

In [17]:
my_data.groupby('Support Function').describe()

Number                          Created         \
                       count unique          top freq   count unique   
Support Function                                                       
Enrollment Issue          78     78  INC27129813    1      78     78   
Failure                   29     29  INC27129200    1      29     29   
Security Policy           19     19  INC27121119    1      19     19   
Security Policy Issue     60     60  INC27146059    1      60     60   
Sychronization Issue      24     24  INC27163035    1      24     24   
Unenrollment               1      1  INC27197684    1       1      1   
Unenrollment Issues        1      1  INC27132927    1       1      1   
Usage Question           228    228  INC27197224    1     228    228   

                                                Caller         \
                                       top freq  count unique   
Support Function                                                
Enrollment Issue       2019-04-19 09:55:51    1     78     77   
Failure                2019-04-17 08:04:18    1     29     29   
Security Policy        2019-04-16 10:04:10    1     19     19   
Security Policy Issue  2019-04-16 11:55:54    1     60     60   
Sychronization Issue   2019-04-16 12:43:39    1     24     24   
Unenrollment           2019-04-22 16:16:28    1      1      1   
Unenrollment Issues    2019-04-17 01:38:55    1      1      1   
Usage Question         2019-04-16 11:58:19    1    228    217   

                                            User ID                      \
                                   top freq   count unique     top freq   
Support Function                                                          
Enrollment Issue       Michael Raynoha    2      78     77  585510    2   
Failure                   Hamid Musavi    1      29     29  592894    1   
Security Policy           John Ruppert    1      19     19  602522    1   
Security Policy Issue   Edward Wilhelm    1      60     60  553090    1   
Sychronization Issue   Daniel Langlais    1      24     24  590615    1   
Unenrollment              Julia Rozman    1       1      1  528068    1   
Unenrollment Issues       Clyde Burton    1       1      1  574766    1   
Usage Question             Frank Brody    2     228    217  601996    2   

                      Short description         \
                                  count unique   
Support Function                                 
Enrollment Issue                     78     75   
Failure                              29     28   
Security Policy                      19     16   
Security Policy Issue                60     58   
Sychronization Issue                 24     24   
Unenrollment                          1      1   
Unenrollment Issues                   1      1   
Usage Question                      228    212   

                                                                               \
                                                                     top freq   
Support Function                                                                
Enrollment Issue           I: Microsoft Authenticator questions - No ETR    4   
Failure                    I: Microsoft Authenticator questions - No ETR    2   
Security Policy           MDM - Security Policy Blocked - Please unblock    3   
Security Policy Issue                              MDM | Security Policy    2   
Sychronization Issue            Intune company portal - upgrade failure     1   
Unenrollment                              MDM | removing previous device    1   
Unenrollment Issues    #1 CMH 04/19/2019 | Microsoft Authenticator Is...    1   
Usage Question             I: Microsoft Authenticator questions - No ETR    4   

                      Description         \
                            count unique   
Support Function                           
Enrollment Issue               78     78   
Failure                        29     29   
Security Policy         

In [18]:
SF.describe()

count                440
unique                 8
top       Usage Question
freq                 228
Name: Support Function, dtype: object

**Conclusion:**

**8 different categories:**

*   Enrollment Issue - 78
*   Failure	- 29
*   Security Policy (Security Policy + Security Policy Issue)	 - 79   
*   Sychronization Issue	- 24
*   Unenrollment - 2
*  **Usage Question - 228**





**Configuration item**

In [0]:
CI = my_data['Configuration item']

In [22]:
CI.head()

0            MDM-HD Resolvable
1    Mobile Apps-HD Resolvable
2               MDM-Escalation
3               MDM-Escalation
4            MDM-HD Resolvable
Name: Configuration item, dtype: object

In [23]:
CI.describe()

count                   440
unique                    5
top       MDM-HD Resolvable
freq                    249
Name: Configuration item, dtype: object

In [25]:
my_data.groupby('Configuration item').describe()

Number                          Created         \
                           count unique          top freq   count unique   
Configuration item                                                         
MDM-Escalation               106    106  INC27190301    1     106    106   
MDM-HD Resolvable            249    249  INC27129813    1     249    249   
MDM-Tier 3                    48     48  INC27119504    1      48     48   
Mobile Apps-Escalation         7      7  INC27119678    1       7      7   
Mobile Apps-HD Resolvable     30     30  INC27119096    1      30     30   

                                                    Caller         \
                                           top freq  count unique   
Configuration item                                                  
MDM-Escalation             2019-04-16 16:35:20    1    106    105   
MDM-HD Resolvable          2019-04-19 09:55:51    1    249    239   
MDM-Tier 3                 2019-04-17 08:04:18    1     48     48   
Mobile Apps-Escalation     2019-04-17 08:50:07    1      7      7   
Mobile Apps-HD Resolvable  2019-04-16 11:58:19    1     30     30   

                                                 User ID                      \
                                        top freq   count unique     top freq   
Configuration item                                                             
MDM-Escalation             Helene Berkholcs    2     106    105  579707    2   
MDM-HD Resolvable             Jeffrey Freel    2     249    239  539414    2   
MDM-Tier 3                     Hamid Musavi    1      48     48  587887    1   
Mobile Apps-Escalation          Brian Frank    1       7      7  598240    1   
Mobile Apps-HD Resolvable     Thomas Arthur    1      30     30  601996    1   

                          Short description         \
                                      count unique   
Configuration item                                   
MDM-Escalation                          106    101   
MDM-HD Resolvable                       249    237   
MDM-Tier 3                               48     44   
Mobile Apps-Escalation                    7      6   
Mobile Apps-HD Resolvable                30     27   

                                                                              \
                                                                         top   
Configuration item                                                             
MDM-Escalation             FW: [External] WARNING: Your device is out of ...   
MDM-HD Resolvable              I: Microsoft Authenticator questions - No ETR   
MDM-Tier 3                    MDM - Security Policy Blocked - Please unblock   
Mobile Apps-Escalation                              authenticator app usage    
Mobile Apps-HD Resolvable                       authenticator app questions    

                               Description         \
                          freq       count unique   
Configuration item                                  
MDM-Escalation               3         106    106   
MDM-HD Resolvable            7         249    247   
MDM-Tier 3                   3          48     48   
Mobile Apps-Escalation       2           7      7   
Mobile Apps-HD Resolvable    2          30     29   

                                                                              \
                                                                         top   
Configuration item                                                             
MDM-Escalation             received from: Rosica_IV@bah.com\n\nI did not ...   
MDM-HD Resolvable          the client is being asked to setup the microso...   
MDM-Tier 3                 Employee ID #: 592894\nEmployee Name: Douglas ...   
Mobile Apps-Escalation     client got an email notice that his device is ...   
Mobile Apps-HD Resolvable      client is trying to do the authenticator app    

                               Support Function                              \
    

**Conclusion:**
*   MDM-Escalation	- 106
*   **MDM-HD Resolvable - 249**
*   MDM-Tier 3 - 48
*   Mobile Apps-Escalation - 7
*  Mobile Apps-HD Resolvable - 30







**Incident State**

In [0]:
IS = my_data['Incident state']

In [28]:
CI.head()

0            MDM-HD Resolvable
1    Mobile Apps-HD Resolvable
2               MDM-Escalation
3               MDM-Escalation
4            MDM-HD Resolvable
Name: Configuration item, dtype: object

In [29]:
CI.describe()

count                   440
unique                    5
top       MDM-HD Resolvable
freq                    249
Name: Configuration item, dtype: object

In [30]:
my_data.groupby('Incident state').describe()

Number                          Created         \
                    count unique          top freq   count unique   
Incident state                                                      
Active                  9      9  INC27134722    1       9      9   
Awaiting Change         9      9  INC27193907    1       9      9   
Awaiting Evidence      15     15  INC27141111    1      15     15   
Awaiting User Info     67     67  INC27127115    1      67     67   
Closed                272    272  INC27132108    1     272    272   
Resolved               66     66  INC27197224    1      66     66   
Work Started            2      2  INC27203974    1       2      2   

                                             Caller                          \
                                    top freq  count unique              top   
Incident state                                                                
Active              2019-04-17 08:04:18    1      9      9   John Scarpelli   
Awaiting Change     2019-04-18 10:16:19    1      9      9     Paul D'Angio   
Awaiting Evidence   2019-04-16 07:07:49    1     15     15     Hamid Musavi   
Awaiting User Info  2019-04-16 08:00:33    1     67     66    Leslie Palmer   
Closed              2019-04-16 11:58:19    1    272    262    Jagjot Wadhwa   
Resolved            2019-04-22 08:02:49    1     66     66     Laura Layton   
Work Started        2019-04-23 09:09:54    1      2      2  Alexander Carey   

                        User ID                     Short description         \
                   freq   count unique     top freq             count unique   
Incident state                                                                 
Active                1       9      9  511682    1                 9      9   
Awaiting Change       1       9      9  535037    1                 9      9   
Awaiting Evidence     1      15     15  543919    1                15     15   
Awaiting User Info    2      67     66  589562    2                67     66   
Closed                2     272    262  534911    2               272    251   
Resolved              1      66     66  574766    1                66     65   
Work Started          1       2      2  544788    1                 2      2   

                                                                            \
                                                                  top freq   
Incident state                                                               
Active              The client states that he can't access his mai...    1   
Awaiting Change         I: Microsoft Authenticator questions - No ETR    1   
Awaiting Evidence                MDM - android - not receiving emails    1   
Awaiting User Info  #1 CG 4/18/2019 FW: Your mobile device has bee...    2   
Closed                  I: Microsoft Authenticator questions - No ETR   10   
Resolved                                                      intune     2   
Work Started        Unable to access Outlook App using MDM on his ...    1   

                   Description         \
                         count unique   
Incident state                          
Active                       9      9   
Awaiting Change              9      9   
Awaiting Evidence           15     15   
Awaiting User Info          67     67   
Closed                     272    270   
Resolved                    66     66   
Work Started                 2      2   

                                                                            \
                                                                  top freq   
Incident state                                                               
Active              received from: gilmer_graham@bah.com\n\nHello,...    1   
Awaiting Change     Employee ID #: 600976\nEmployee Name: Tina Con...    1   
Awaiting Evidence   Employee ID #: 592894\nEmployee Name: Douglas ...    1   
Awaiting User Info  forwarded by: herron_travis@bah.com\n\nHelp De...    1  

**Conclusion**

There are  5 different categories:
*   Awaiting Evidance - 15
*   Awaiting Use Info - 67
*   **Closed - 272**
*   Resolved - 66
*   Work - 2





**Assignemnt Group**

In [0]:
AG = my_data['Assignment group']

In [32]:
AG.head()

0        BAH Service Desk
1        BAH Service Desk
2    BAH BSO ESC Wireless
3    BAH BSO ESC Wireless
4        BAH Service Desk
Name: Assignment group, dtype: object

In [33]:
AG.describe()

count                  440
unique                   7
top       BAH Service Desk
freq                   213
Name: Assignment group, dtype: object

In [36]:
my_data.groupby('Assignment group').describe()

Number                          Created  \
                                   count unique          top freq   count   
Assignment group                                                            
BAH BSO ESC Wireless                 110    110  INC27190301    1     110   
BAH IS EUE Endpoint Engineering       48     48  INC27119504    1      48   
BAH IS Enterprise Email                1      1  INC27134885    1       1   
BAH Service Desk                     213    213  INC27129813    1     213   
BAH Service Desk Pending Closure      16     16  INC27183773    1      16   
BAH Service Desk Pending Customer     48     48  INC27173126    1      48   
BAH Service Desk Self-Service          4      4  INC27146199    1       4   

                                                                   Caller  \
                                  unique                  top freq  count   
Assignment group                                                            
BAH BSO ESC Wireless                 110  2019-04-16 16:35:20    1    110   
BAH IS EUE Endpoint Engineering       48  2019-04-17 08:04:18    1     48   
BAH IS Enterprise Email                1  2019-04-17 08:30:07    1      1   
BAH Service Desk                     213  2019-04-16 11:58:19    1    213   
BAH Service Desk Pending Closure      16  2019-04-17 06:16:16    1     16   
BAH Service Desk Pending Customer     48  2019-04-16 12:11:21    1     48   
BAH Service Desk Self-Service          4  2019-04-17 15:40:20    1      4   

                                                                 User ID  \
                                  unique                top freq   count   
Assignment group                                                           
BAH BSO ESC Wireless                 108     Mohammad Firoz    2     110   
BAH IS EUE Endpoint Engineering       48       Hamid Musavi    1      48   
BAH IS Enterprise Email                1       Daniel Alton    1       1   
BAH Service Desk                     205      Jeffrey Freel    2     213   
BAH Service Desk Pending Closure      16     Robert Allegar    1      16   
BAH Service Desk Pending Customer     47      Leslie Palmer    2      48   
BAH Service Desk Self-Service          4  Hasnain Chowdhury    1       4   

                                                      Short description  \
                                  unique     top freq             count   
Assignment group                                                          
BAH BSO ESC Wireless                 108  598240    2               110   
BAH IS EUE Endpoint Engineering       48  587887    1                48   
BAH IS Enterprise Email                1  023147    1                 1   
BAH Service Desk                     205  534911    2               213   
BAH Service Desk Pending Closure      16  603296    1                16   
BAH Service Desk Pending Customer     47  589562    2                48   
BAH Service Desk Self-Service          4  602600    1                 4   

                                          \
                                  unique   
Assignment group                           
BAH BSO ESC Wireless                 104   
BAH IS EUE Endpoint Engineering       44   
BAH IS Enterprise Email                1   
BAH Service Desk                     198   
BAH Service Desk Pending Closure      16   
BAH Service Desk Pending Customer     47   
BAH Service Desk Self-Service          4   

                                                                                      \
                                                                                 top   
Assignment group                                                                       
BAH BSO ESC Wireless               FW: [External] WARNING: Your device is out of ...   
BAH IS EUE Endpoint Engineering       MDM - Security Policy Blocked - Please unblock   
BAH IS Enterprise Email                       outlook app | unable to get signed in    
BAH Service

**Conclusion**

There are 7 unique categories:
*   BAH BSO ESC Wireless - 110
*   BAH IS EUE Endpoint Engineering - 48
*   BAH IS Enterprise Email - 1
*   **BAH Service Desk - 213**
*   BAH Service Desk Pending Closure - 16
*   BAH Service Desk Pending Customer - 48
*   BAH Service Desk Self-Service - 4







**Contact Type**

In [0]:
CT = my_data['Contact type']

In [39]:
CT.head()

0    Phone
1    Phone
2    Phone
3    Phone
4    Phone
Name: Contact type, dtype: object

In [40]:
CT.describe()

count       440
unique        6
top       Phone
freq        309
Name: Contact type, dtype: object

In [41]:
my_data.groupby('Contact type').describe()

Number                          Created         \
              count unique          top freq   count unique   
Contact type                                                  
Chat             18     18  INC27197224    1      18     18   
Email           101    101  INC27132108    1     101    101   
External          1      1  INC27128223    1       1      1   
Phone           309    309  INC27129813    1     309    309   
Self-service      9      9  INC27147040    1       9      9   
Walk-in           2      2  INC27135304    1       2      2   

                                       Caller                                \
                              top freq  count unique               top freq   
Contact type                                                                  
Chat          2019-04-16 10:59:24    1     18     18   Benson Ayorinde    1   
Email         2019-04-17 18:34:30    1    101     99     Leslie Palmer    2   
External      2019-04-16 15:41:00    1      1      1   William Preston    1   
Phone         2019-04-17 08:04:18    1    309    287  Helene Berkholcs    3   
Self-service  2019-04-18 10:16:19    1      9      9        John Guris    1   
Walk-in       2019-04-17 08:50:07    1      2      2    Mokhtar Zitoun    1   

             User ID                     Short description         \
               count unique     top freq             count unique   
Contact type                                                        
Chat              18     18  596489    1                18     18   
Email            101     99  593399    2               101     95   
External           1      1  542144    1                 1      1   
Phone            309    287  579707    3               309    284   
Self-service       9      9  582075    1                 9      9   
Walk-in            2      2  801080    1                 2      2   

                                                                      \
                                                            top freq   
Contact type                                                           
Chat          Customer was unable to login to Outlook on And...    1   
Email         FW: [External] WARNING: Your device is out of ...    5   
External      Assistance with Intune BAH apps installation i...    1   
Phone             I: Microsoft Authenticator questions - No ETR   12   
Self-service  #1 aM 4-23-19 I am currently listed as an "on-...    1   
Walk-in                        BAH email issue in Android phone    1   

             Description         \
                   count unique   
Contact type                      
Chat                  18     18   
Email                101    101   
External               1      1   
Phone                309    306   
Self-service           9      9   
Walk-in                2      2   

                                                                      \
                                                            top freq   
Contact type                                                           
Chat          Employee ID #: 592894\nEmployee Name: Douglas ...    1   
Email         received from: winter_heather@bah.com\n\nHi,\n...    1   
External      Assistance with Intune BAH apps installation i...    1   
Phone         the client is being asked to setup the microso...    3   
Self-service  I am currently listed as an "on-premesis user"...    1   
Walk-in                        BAH email issue in Android phone    1   

             Configuration item                                      \
                          count unique                     top freq   
Contact type                                                          
Chat                         18      4       MDM-HD Resolvable    8   
Email                       101      4       MDM-HD Resolvable   64   
External                      1      1              MDM-Tier 3    1   
Phone                       309      5       MDM-HD Resolvable  174   


**Conclusion**

There are 6 types in total:
*   Chat  -  18
*   Email  - 101
*   External   - 1
*   **Phone  - 309**
*   Self-service  - 9
*   Walk-in  - 2







**Description**

In [0]:
Description = my_data['Description']

In [43]:
Description.head(15)

0     Employee ID #: 544788\nEmployee Name: Alexande...
1     Employee ID #: 556147\nEmployee Name: Jane Bla...
2     Employee ID #: 584025\nEmployee Name: Thomas A...
3     Employee ID #: 599061\nEmployee Name: Dan Hanc...
4     unable to access emails on phone\nadvised to i...
5     Issue Description: ­­­­\nThe client states tha...
6     Mr. Ngoc Chu is reporting when he logs onto OU...
7     mam device in not in policy \nclient keeps get...
8     Client is needing assistance with the authenti...
9     Ms. Judith Mayer is needing assistance with th...
10    Ms. Cammela Budram-Dawkins is reporting that s...
11    Employee ID #: 008799\nEmployee Name: Jeffrey ...
12    Employee ID #: 528068\nEmployee Name: Julia Ro...
13    User received an email saying that his device ...
14    Jennifer Vincent : I think I need to update/re...
Name: Description, dtype: object

**Conclusion:**

We may be able to do NLP on Description part, however, the content itself is a big mess.

**Questions:**

*   Number - How many distinct ticket?
*   Created - See the frequency distribution within one week?
*   Caller/User_ID - Who has the most question? If we could group people to each department, we potentially could know which department has the most amounts of data, and then we could give them a specific training? Or help them solve the problem from root?
*   Short Description - Didn't contain too many details, also too various to be categorized.



